docs/storage-surface.rst

```diff
@@ -0,0 +1,673 @@
+
+.. code-block:: python
+
+   >>> bucket.delete(force=True)
+
+as above.
+
+To make updates to the bucket use
+:meth:`Bucket.patch() <gcloud.storage.bucket.Bucket.patch>`
+
+.. code-block:: python
+
+   >>> bucket.versioning_enabled = True
+   >>> bucket.patch()
+
+If there are no updates to send, an exception will occur
+
+.. code-block:: python
+
+   >>> bucket.patch()
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   ValueError: No updates to send.
+
+In total, the properties that can be updated are
+
+.. code-block:: python
+
+   >>> bucket.cors = [
+   ...     {
+   ...       'origin': ['http://example.appspot.com'],
+   ...       'responseHeader': ['Content-Type'],
+   ...       'method': ['GET', 'HEAD', 'DELETE'],
+   ...       'maxAgeSeconds': 3600,
+   ...     }
+   ... ]
+   >>> bucket.lifecycle = [
+   ...     {
+   ...         'action': {'type': 'Delete'},
+   ...         'condition': {'age': 365},
+   ...     },
+   ... ]
+   >>> bucket.location = 'ASIA'
+   >>> bucket.logging = {
+   ...     'logBucket': 'bucket-name',
+   ...     'logObjectPrefix': 'foo/',
+   ... }
+   >>> bucket.storage_class = 'DURABLE_REDUCED_AVAILABILITY'
+   >>> bucket.versioning_enabled = True
+   >>> bucket.website = {
+   ...     'mainPageSuffix': 'index.html',
+   ...     'notFoundPage': '404.html',
+   ... }
+
+In general, many of these properties are optional and will not need to be
+used (or changed from the defaults).
+
+In addition, a bucket has several read-only properties
+
+.. code-block:: python
+
+   >>> bucket.etag
+   u'CAI='
+   >>> bucket.id
+   u'bucket-name'
+   >>> bucket.metageneration
+   2L
+   >>> bucket.name
+   u'bucket-name'
+   >>> bucket.owner
+   {u'entity': u'project-owners-111111'}
+   >>> bucket.project_number
+   u'111111'
+   >>> bucket.self_link
+   u'https://www.googleapis.com/storage/v1/b/bucket-name'
+   >>> bucket.time_created
+   u'2015-01-01T12:00:00.000Z'
+
+See `buckets`_ specification for more details.
+
+Other data -- namely `access control`_ data -- is associated with buckets,
+but is handled through ``Bucket.acl``.
+
+.. _buckets: https://cloud.google.com/storage/docs/json_api/v1/buckets
+.. _access control: https://cloud.google.com/storage/docs/access-control
+
+.. note::
+  **BREAKING THE FOURTH WALL**: Note that ``storage.buckets.update`` is
+  absent. This doesn't seem necessary to implement given the presence of
+  ``patch``.
+
+Manage Blobs
+------------
+
+Interacting with blobs requires an associated bucket. Either methods on an
+explicit :class:`Bucket <gcloud.storage.bucket.Bucket>` instance can be used
+or the default bucket can be used via methods in the ``storage`` package.
+
+To retrieve a blob with a bucket
+
+.. code-block:: python
+
+   >>> bucket.get_blob('blob-name')
+   <Blob: bucket-name, blob-name>
+
+or from the package via the default bucket
+
+.. code-block:: python
+
+   >>> storage.get_blob('blob-name')
+   <Blob: default-bucket-name, blob-name>
```

Comment:
Yes. It's debatable if it's necessary, but essentially the goal is to mirror all the relevant methods on `Bucket`, but allow use of the default bucket. (This was, people can just treat Cloud Storage like a filesystem and only worry about files.)